cpp/src/arrow/compute/kernels/vector_selection_take_internal.cc

```diff
@@ -357,15 +481,22 @@ template <typename IndexCType, typename ValueBitWidthConstant,
 struct FixedWidthTakeImpl {
   static constexpr int kValueWidthInBits = ValueBitWidthConstant::value;
 
-  static Status Exec(KernelContext* ctx, const ArraySpan& values,
+  static Status Exec(KernelContext* ctx, const ValuesSpan& values,
                      const ArraySpan& indices, ArrayData* out_arr, int64_t factor) {
 #ifndef NDEBUG
-    int64_t bit_width = util::FixedWidthInBits(*values.type);
+    int64_t bit_width = util::FixedWidthInBits(*values.type());
     DCHECK(WithFactor::value || (kValueWidthInBits == bit_width && factor == 1));
     DCHECK(!WithFactor::value ||
            (factor > 0 && kValueWidthInBits == 8 &&  // factors are used with bytes
             static_cast<int64_t>(factor * kValueWidthInBits) == bit_width));
 #endif
+    return values.is_chunked()
+               ? ChunkedExec(ctx, values.chunked_array(), indices, out_arr, factor)
+               : Exec(ctx, values.array(), indices, out_arr, factor);
+  }
+
+  static Status Exec(KernelContext* ctx, const ArraySpan& values,
```

Comment:
Perhaps call this `ContiguousExec` similar to `ChunkedExec` below?

Reply:
The "exec" and "chunked exec" dual seems to be all over the place. Adding `Contiguous` here would introduce new terminology.
https://github.com/apache/arrow/blob/cc3c868aea7317a58447658f1c165ad352cd4865/cpp/src/arrow/compute/kernel.h#L600-L605

Reply:
Related: https://github.com/apache/arrow/pull/41700#discussion_r1641239721